Gabriel da Silva Carvalho 

Matricula - 0050013382

base de dados escolhida: http://archive.ics.uci.edu/ml/datasets/Wine+Quality

Link da pasta do google drive - https://drive.google.com/drive/u/0/folders/1YnJLHFIMjaTKo-8A40PL4OLsYG1xM3f2

2.1 - Importar os modulos python para machine learn e carregar o arquivo

** Sugestão se for do tipo xlsx, usar o read_excel do pandas import pandas as pd dt = pd.read_excel("meuarquivo.xlsx")

In [ ]:
import pandas as pd
import numpy as np

In [90]:
arquivo_wine = pd.read_excel("/content/wine.xlsx")

In [91]:
arquivo_wine.describe()

fixed acidity  volatile acidity  ...      alcohol      quality
count    1599.000000       1599.000000  ...  1599.000000  1599.000000
mean        8.319637          0.527821  ...    10.422983     5.636023
std         1.741096          0.179060  ...     1.065668     0.807569
min         4.600000          0.120000  ...     8.400000     3.000000
25%         7.100000          0.390000  ...     9.500000     5.000000
50%         7.900000          0.520000  ...    10.200000     6.000000
75%         9.200000          0.640000  ...    11.100000     6.000000
max        15.900000          1.580000  ...    14.900000     8.000000

[8 rows x 12 columns]

2.2 - Exibir os primeiros registros desta tabela.

In [92]:
arquivo_wine.head()

fixed acidity  volatile acidity  citric acid  ...  sulphates  alcohol  quality
0            7.4              0.70         0.00  ...       0.56      9.4        5
1            7.8              0.88         0.00  ...       0.68      9.8        5
2            7.8              0.76         0.04  ...       0.65      9.8        5
3           11.2              0.28         0.56  ...       0.58      9.8        6
4            7.4              0.70         0.00  ...       0.56      9.4        5

[5 rows x 12 columns]

2.3 - De que se trata esse banco de dados? e que tipo de previsão pode ser feito com ele?

Resposta: O dataset tem como objetivo examinar 11 features para conseguir chegar no valor numérico de qualidade de cada um desses vinhos.

3.1 - Quantas "features" têm nessa base de dados?

Resposta: 11 Features.

3.2 - Quantas observações têm nessa base de dados?

Resposta: 1599 observações.

In [73]:
feature_cols = ['fixed acidity', 'volatile acidity', 'citric acid', 'residual sugar', 'chlorides', 'free sulfur dioxide', 'total sulfur dioxide', 'density', 'pH', 'sulphates', 'alcohol']

In [74]:
X = arquivo_wine[feature_cols]

In [75]:
y = arquivo_wine['quality']

4.1 - Faça uma previsão, usando o algoritmo de LogisticRegression.

In [88]:
X.shape

(1599, 11)

In [89]:
y.shape

(1599,)

In [94]:
from sklearn.linear_model import LogisticRegression
logreg = LogisticRegression()
logreg.fit(X, y)
logreg.predict(X)

/usr/local/lib/python3.6/dist-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


array([5, 5, 5, ..., 6, 5, 6])

In [95]:
y_pred = logreg.predict(X)
len(y_pred)

1599

In [96]:
from sklearn import metrics
print(metrics.accuracy_score(y, y_pred))

0.5797373358348968


5.0 - Usando o método de avaliação de acurácia (Treinar e testar na base de dados inteira/Train test entire model), Calcular a acurácia de cada um dos 3 métodos abaixo.

5.1 - Acurácia usando o algoritmo de KNN (com 1 vizinho, k=1)

In [97]:
knn = KNeighborsClassifier(n_neighbors=1)
knn.fit(X, y)
y_pred = knn.predict(X)
print(metrics.accuracy_score(y, y_pred))

1.0


5.2 - Acurácia usando o algoritmo de KNN (com 5 vizinho, k=5)

In [98]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=5)
knn.fit(X, y)
y_pred = knn.predict(X)
print(metrics.accuracy_score(y, y_pred))

0.6704190118824265


5.3 - Acurácia usando o algoritmo de LogisticRegression

In [99]:
from sklearn.linear_model import LogisticRegression
logreg = LogisticRegression()
logreg.fit(X, y)
logreg.predict(X)
y_pred = logreg.predict(X)
len(y_pred)
print(metrics.accuracy_score(y, y_pred))

0.5797373358348968


/usr/local/lib/python3.6/dist-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


In [101]:
y_pred = logreg.predict(X_test)
print(metrics.accuracy_score(y_test, y_pred))

0.5671875


5.4 - De acordo com seus resultados anteriores, qual dos 3 métodos é mais eficiente? 

Resposta: KNN-1